test/tests/parallel/set3/mta/dsn.bats

```diff
@@ -0,0 +1,88 @@
+load "${REPOSITORY_ROOT}/test/helper/setup"
+load "${REPOSITORY_ROOT}/test/helper/common"
+
+BATS_TEST_NAME_PREFIX='[DSN] '
+CONTAINER_NAME='dms-test_dsn'
```

Comment:
At a glance, I think the issue could be from here.

- The test goes straight into test cases, no `setup_file()`.
- The `teardown()` default should handle this inbetween without issue though, so a shared `CONTAINER_NAME` shouldn't be an issue due to sequential test run 🤷‍♂️ (_parallel is only relevant to separate test files concurrently, not their individual test cases within_)
- But given that the 2nd and 3rd test cases are the ones that fail, it does seem like the containers may have failed to start correctly, thus the attempt to send mail to port 25 fails.